dns.tf

```diff
@@ -0,0 +1,29 @@
+resource "digitalocean_domain" "top_level_domains" {
+    for_each = toset(var.top_level_domains)
+    name = each.value
+}
+
+resource "digitalocean_record" "a_records" {
+  for_each = toset(var.top_level_domains)
+  domain = each.value
+  type   = "A"
+  ttl = 60
+  name   = "@"
+  value  = digitalocean_loadbalancer.ingress_load_balancer.ip
+  depends_on = [
+    digitalocean_domain.top_level_domains,
```

Comment:
Remove this dependency and instead specify the attribute [`domain`](https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/resources/record#domain) that would be set to `digitalocean_domain.top_level_domains[each.key].id` (or something like this) as per the examples in https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/resources/record#example-usage

Reply:
The attribute `domain` is already set: https://github.com/jenkins-infra/digitalocean/pull/63/files#diff-d160bcd3ef13731ac5d7aa43f0c1e76f5635c0a2284a2af2dd35c86a124aee5dR8
And the dependency is needed as the record can't be created if the domain isn't.

Reply:
The examples show that the id of the terraofrm resource is needed for the `domain` attribute (which, by the way, would define the depdnency explictly)